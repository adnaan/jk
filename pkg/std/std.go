package std

import (
	"log"

	"github.com/dlespiau/jk/pkg/__std"

	flatbuffers "github.com/google/flatbuffers/go"
)

// Execute parses a message from v8 and execute the corresponding function.
func Execute(msg []byte) []byte {
	message := __std.GetRootAsMessage(msg, 0)

	union := flatbuffers.Table{}
	if !message.Args(&union) {
		log.Fatal("could not parse Message")
	}

	switch message.ArgsType() {
	case __std.ArgsWriteArgs:
		args := __std.WriteArgs{}
		args.Init(union.Bytes, union.Pos)
		write(args.Value(), string(args.Path()), args.Type())
		return nil
	default:
		log.Fatalf("unknown Message (%d)", message.ArgsType())
		return nil
	}
}
