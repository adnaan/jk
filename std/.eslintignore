flatbuffers.js
flatbuffers.d.ts
__std*.js
__std*.ts
