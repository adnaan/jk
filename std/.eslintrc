{
  "extends": "airbnb-base",
  "plugins": ["eslint-plugin-jest"],
  "globals": {
    "V8Worker2": true
  },
  "settings": {
     "import/resolver": {
        "node": {
          "paths": ["."]
        }
     }
  },
  "env": {
    "jest/globals": true
  },
  "rules": {
    "no-continue": 0,
    "prefer-const": ["error", {"destructuring": "all"}],
    "import/prefer-default-export": 0,
    "import/no-unresolved": [
      2,
      {
        "ignore": [
          "^@jkcfg/std$"
        ]
      }
    ],
    "indent": ["error", 2, {"SwitchCase": 0, "CallExpression": {"arguments": "first"}}],
    "no-restricted-syntax": ["error", "ForInStatement", "LabeledStatement", "WithStatement"],
    "no-use-before-define": ["error", { "functions": false }]
  }
}
