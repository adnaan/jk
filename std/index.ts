export { log } from './log';
export {
  stdout,
  Format,
  Overwrite,
  write,
  print,
} from './write';
export { Encoding, read, stdin } from './read';
export { parse, stringify } from './parse';
