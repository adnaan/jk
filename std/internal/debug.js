import { RPC } from './rpc';

export function debug(...args) {
  return RPC("debug", ...args);
}
