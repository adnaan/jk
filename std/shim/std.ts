function log(value: any): void { };

enum Format {
    JSON = 1,
    YAML,
    Raw,
}

interface WriteOptions {
  format?: Format,
  indent?: number,
  override?: boolean,
}

function write(value: any, path: string, options: WriteOptions): void { };

enum Encoding {
  Bytes,
  String,
  JSON,
}

interface ReadOptions {
  format: Format,
  encoding: Encoding,
}

function read(path: string, options: ReadOptions): Promise<ArrayBuffer | string | object> { return Promise.resolve({}) };

export default {
  log, Format, write, read,
};
