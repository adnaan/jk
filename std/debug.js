import { valueFromUTF8Bytes } from './internal/data';
import { RPC } from './internal/rpc';

export function echo(...args) {
  return RPC('debug.echo', ...args).then(valueFromUTF8Bytes);
}
