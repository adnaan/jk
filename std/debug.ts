/**
 * @module std/debug
 *
 * debug exists to help with testing the runtime.
 */

import { valueFromUTF8Bytes } from './internal/data';
import { RPC, RPCSync } from './internal/rpc';

export function echo(...args: any[]): Promise<any[]> {
  return RPC('debug.echo', ...args).then(valueFromUTF8Bytes);
}

export function echoSync(...args: any[]): any[] {
  return valueFromUTF8Bytes(RPCSync('debug.echo', ...args));
}
