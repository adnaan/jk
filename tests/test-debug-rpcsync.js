import { echoSync } from '@jkcfg/std/debug';
import { print } from '@jkcfg/std';

print(echoSync());

const arr = new Uint8Array(new ArrayBuffer(3));
arr[0] = 1;
arr[1] = 2;
arr[2] = 3;

print(echoSync(65, arr, 'string', { object: 'object' }));
