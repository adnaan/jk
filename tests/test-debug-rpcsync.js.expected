[]
[
  65,
  [
    1,
    2,
    3
  ],
  "string",
  {
    "object": "object"
  }
]
